### DISCUSSION TITLE
SetNickName & RoleUpdate for "interaction" Feature Request

### DISCUSSION THREAD

**Discussion by ramazansancar** [discordjs/discord.js#8354] [Q&A]
### Which package is the feature request for?

builders

### Feature

I get it from the user with the "/"(slash) command with interaction. And if the code entered by the user is correct with the verification system I wrote, I want to assign roles and change usernames.
However, interaction does not have the authority to change the username and assign a role. What can I do for it?

### Ideal solution or implementation

Adding authorizations to the interaction and opening the scope

### Alternative solutions or implementations

This can be achieved with the Rest Api, and there should be no OAuth requirement for role assignment.

### Other context

Not now

**Comment by ramazansancar** [comment]
Why can we only do this with OAuth2?

**Reply by Jiralite** [reply]
Giving a role or updating a member's nickname requires the necessary permissions. If your application's bot is not in the server, you cannot do it. If it is, you need permission. You can do this with OAuth2 as well.

**Reply by ramazansancar** [reply]
> Giving a role or updating a member's nickname requires the necessary permissions. If your application's bot is not in the server, you cannot do it. If it is, you need permission. You can do this with OAuth2 as well.

@Jiralite I added the Discord Bot to the server. And I'm aiming to update their names to give them roles by passing them through a verification system. But for Node.js, I couldn't provide this in the bot. Previously, when I got permission with OAuth2, I could add it to the server I wanted and give it the role. However, I could not find any documentation or example on how to do this in a bot running on the Cli.
